src/storage/core.rs

```diff
@@ -897,29 +901,39 @@ where
     async fn mark_all_as_deleted_closed(&self, key: &K) -> Result<u64> {
         let safe = self.inner.safe.write().await;
         let mut blobs = safe.blobs.write().await;
-        let entries_closed_blobs = blobs
-            .iter_mut()
-            .map(|b| b.mark_all_as_deleted(key))
-            .collect::<FuturesUnordered<_>>();
-        let total = entries_closed_blobs
-            .filter_map(|result| match result {
-                Ok(count) => count,
-                Err(error) => {
-                    warn!("failed to delete records: {}", error);
-                    None
+        let count = AtomicU64::new(0);
```

Comment:
IMHO, original implementation of this function without Atomics was clearer and simpler

Reply:
You right, since we don't need ownership here I'll return old function